lib/fluent/parser.rb

```diff
@@ -548,7 +552,26 @@ def initialize
       def configure(conf)
         super
 
-        @regexp = @with_priority ? REGEXP_WITH_PRI : REGEXP
+        @time_parser_rfc3164 = @time_parser_rfc5424 = nil
+        @regexp = case @message_format
+                  when :rfc3164
+                    class << self
+                      alias_method :parse, :parse_plain
+                    end
+                    @with_priority ? REGEXP_WITH_PRI : REGEXP
+                  when :rfc5424
+                    class << self
+                      alias_method :parse, :parse_plain
+                    end
+                    REGEXP_RFC5424
```

Comment:
Ah! Your suggestion means that we should set `@time_format` in the following situation?

```ruby
@time_format = @rfc5424_time_format unless @time_format
```

It is reasonable for me.